### PR TITLE
S3 Static Website sample using Pulumi

### PR DESCRIPTION
**Pull request by HarshCasper**

### CODE REVIEW

File: s3-static-website/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+const mime = require("mime");
+
+// Create a bucket and expose a website index document
+let siteBucket = new aws.s3.Bucket("s3-website-bucket", {
+    website: {
+        indexDocument: "index.html",
+    },
+});
+
+let siteDir = "www"; // directory for content files
```
Comment (whummer) [inline s3-static-website/index.js:14]:
```suggestion
const siteDir = "www"; // directory for content files
```

File: s3-static-website/Pulumi.localstack.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+config:
```
Comment (whummer) [inline s3-static-website/Pulumi.localstack.yaml:1]: Same as in the other PR, perhaps we can avoid checking in this file to the repo?

**Review by whummer — APPROVED** [review]
Nice sample 👍

File: s3-static-website/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+pulumilocal up
+```
+
+After the Pulumi stack has been deployed, you should be able to check the available S3 buckets via:
+
+```bash
+awslocal s3 ls
+```
+
+You can access the S3 website by using the bucket name in the following format: `http://<BUCKET_NAME>.s3-website.localhost.localstack.cloud:4566`. Since the endpoint is configured to use `localhost.localstack.cloud`, no real AWS resources have been created.
```
Comment (bentsku) [inline s3-static-website/README.md:55]: nit: from the file above, I think the endpoint used is `localhost:4566` with `aws:s3ForcePathStyle: 'true'`, but I'm not sure this is really important or the point of that sentence.. Just that `localhost.localstack.cloud` does not work with S3 if using virtual host, and usually should not be used with S3. S3 endpoints are a bit complicated 😭

File: s3-static-website/index.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+// For each file in the directory, create an S3 object stored in `siteBucket`
+for (let item of require("fs").readdirSync(siteDir)) {
+    let filePath = require("path").join(siteDir, item);
+    let object = new aws.s3.BucketObject(item, {
+        bucket: siteBucket,                               // reference the s3.Bucket object
+        source: new pulumi.asset.FileAsset(filePath),     // use FileAsset to point to a file
+        contentType: mime.getType(filePath) || undefined, // set the MIME type of the file
```
Comment (bentsku) [inline s3-static-website/index.js:22]: note: this is great! nice use of the automatic mime type that's usually missing.

File: s3-static-website/index.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            ],
+            Resource: [
+                `arn:aws:s3:::${bucketName}/*` // policy refers to bucket name explicitly
+            ]
+        }]
+    };
+}
+
+// Set the access policy for the bucket so all objects are readable
+let bucketPolicy = new aws.s3.BucketPolicy("bucketPolicy", {
```
Comment (bentsku) [inline s3-static-website/index.js:44]: I guess same as Waldemar for the `const` usage:
```suggestion
const bucketPolicy = new aws.s3.BucketPolicy("bucketPolicy", {
```

File: s3-static-website/index.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        indexDocument: "index.html",
+    },
+});
+
+let siteDir = "www"; // directory for content files
+
+// For each file in the directory, create an S3 object stored in `siteBucket`
+for (let item of require("fs").readdirSync(siteDir)) {
+    let filePath = require("path").join(siteDir, item);
+    let object = new aws.s3.BucketObject(item, {
```
Comment (bentsku) [inline s3-static-website/index.js:19]:
```suggestion
    const filePath = require("path").join(siteDir, item);
    const object = new aws.s3.BucketObject(item, {
```

**Review by bentsku — APPROVED** [review]
LGTM! Awesome, thanks for taking care of this! 😄 it looks great, and it's rather simple, that's cool. 

Just one note about the endpoint comment, not sure if it should be deleted or reformulated to not confuse people, but maybe it's also me being a bit weird about S3 endpoints. Nothing blocking!

**Pull request merged**